src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -496,6 +497,15 @@ public override Type[] GetOptionalCustomModifiers()
                 m_signature.GetCustomModifiers(PositionImpl + 1, false);
         }
 
+        public override bool Equals(object? obj) =>
+            ReferenceEquals(this, obj) ||
+                (MetadataUpdater.IsSupported &&
+                    obj is RuntimeParameterInfo pi &&
+                    m_tkParamDef == pi.m_tkParamDef &&
+                    ReferenceEquals(DefiningMethod.DeclaringType, pi.DefiningMethod.DeclaringType));
```

Comment:
Whis does not work for global methods. DeclaringType is going to be null for them, and so we may incorrectly return that two ParameterInfos are equal even though they are from two unrelated modules.